dcos/mesos.py

```diff
@@ -1015,8 +1015,16 @@ def __init__(self, task_id, cmd=None, args=None,
         client = DCOSClient()
         master = get_master(client)
 
-        # Get the URL to the agent running the task.
+        # Get the task and make sure its a container launched by the UCR.
         task_obj = master.task(task_id)
+        if "container" in task_obj.dict():
+            if "type" in task_obj.dict()["container"]:
+                if task_obj.dict()["container"]["type"] != "MESOS":
```

Comment:
lets just use check `if task_obj.dict()["container"]["type"] != "MESOS":` and catch `ValueError`. It's a little cleaner than the nesting `ifs`